components/raftstore-v2/src/operation/command/mod.rs

```diff
@@ -147,19 +206,160 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
                     // on_committed callback if necessary.
                     p.cb.notify_committed();
                 }
-                entry_and_proposals.push((e, proposal));
+                entry_and_proposals.push((e, proposal.map_or_else(Vec::new, |p| p.cb)));
             }
         } else {
-            entry_and_proposals = committed_entries.into_iter().map(|e| (e, None)).collect();
-        }
-        // Note that the `commit_index` and `commit_term` here may be used to
-        // forward the commit index after being restarted. So it must be less
-        // than or equal to persisted index.
-        let commit_index = cmp::min(
-            self.raft_group().raft.raft_log.committed,
-            self.raft_group().raft.raft_log.persisted,
-        );
-        let commit_term = self.raft_group().raft.raft_log.term(commit_index).unwrap();
-        // TODO: schedule apply task
+            entry_and_proposals = committed_entries.into_iter().map(|e| (e, vec![])).collect();
+        }
+        // Unlike v1, v2 doesn't need to persist commit index and commit term. The
+        // point of persist commit index/term of raft apply state is to recover commit
+        // index when the writes to raft engine is lost but writes to kv engine is
```

Comment:
Question: In v1, can entries in ready.committed_entries() be unpersisted?  Otherwise, what does it mean "writes to raft engine is lost?"

Reply:
In raft_log.rs next_entries_since, it seems it should return all committed and persisted entries.

Reply:
> can entries in ready.committed_entries() be unpersisted

Yes. As long as it's committed (persisted in quorum), it will be applied without waiting for being persisted.

Reply:
From the code, it seems even v1 will need the entries be persisted before apply.